node/hack/repair_terra/repair.go

```diff
@@ -0,0 +1,428 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/db"
+	"go.uber.org/zap"
+
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	nodev1 "github.com/certusone/wormhole/node/pkg/proto/node/v1"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	eth_common "github.com/ethereum/go-ethereum/common"
+
+	"github.com/tidwall/gjson"
+	"google.golang.org/grpc"
+)
+
+var (
+	adminRPC   = flag.String("adminRPC", "/run/guardiand/admin.socket", "Admin RPC address")
+	terraAddr  = flag.String("terraProgram", "terra1dq03ugtd40zu9hcgdzrsq6z2z4hwhc9tqk2uy5", "Terra program address")
+	dryRun     = flag.Bool("dryRun", true, "Dry run")
+)
+
+func getAdminClient(ctx context.Context, addr string) (*grpc.ClientConn, error, nodev1.NodePrivilegedServiceClient) {
+	conn, err := grpc.DialContext(ctx, fmt.Sprintf("unix:///%s", addr), grpc.WithInsecure())
+
+	if err != nil {
+		log.Fatalf("failed to connect to %s: %v", addr, err)
+	}
+
+	c := nodev1.NewNodePrivilegedServiceClient(conn)
+	return conn, err, c
+}
+
+type logResponse struct {
+	// "1" if ok, "0" if error
+	Status string `json:"status"`
+	// "OK" if ok, "NOTOK" otherwise
+	Message string `json:"message"`
+	// String when status is "0", result type otherwise.
+	Result json.RawMessage `json:"result"`
+}
+
+func getCurrentHeight() (uint64, error) {
+	client := &http.Client{ 
+		Timeout: time.Second * 5, 
+	}
+
+	// Query and report height and set currentTerraHeight
+	terraWS := "http://localhost:1317"
+	resp, err := client.Get(fmt.Sprintf("%s/blocks/latest", terraWS))
+	if err != nil {
+		return 0, fmt.Errorf("failed to get current height: %w", err)
+	}
+	blocksBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get current height: %w", err)
+	}
+	resp.Body.Close()
+
+	blockJSON := string(blocksBody)
+	latestBlock := gjson.Get(blockJSON, "block.header.height")
+	fmt.Println("current Terra height ", latestBlock.Int())
+	return latestBlock.Uint(), nil
+}
+
+func getSequenceForTxhash(txhash string) (uint64, error) {
+	// fmt.Println("Starting getSequence()...")
+	client := &http.Client{ 
+		Timeout: time.Second * 5, 
+	}
+	url := fmt.Sprintf("https://fcd.terra.dev/cosmos/tx/v1beta1/txs/%s", txhash)
+	// fmt.Println("url", url)
+	resp, err := client.Get(url)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get message: %w", err)
+	}
+	txBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		resp.Body.Close()
+		return 0, fmt.Errorf("failed to read message: %w", err)
+	}
+	resp.Body.Close()
```

Comment:
```suggestion
	resp, err := client.Get(url)
	if err != nil {
		return 0, fmt.Errorf("failed to get message: %w", err)
	}
	defer resp.Body.Close()
	txBody, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		resp.Body.Close()
		return 0, fmt.Errorf("failed to read message: %w", err)
	}
```

defer the close after the initial err check so the response body is actually closed if the second error condition hits.

Reply:
Made the suggested change